openapi-cli/src/test/java/io/ballerina/generators/auth/HttpOrOAuthTests.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.generators.auth;
+
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.VariableDeclarationNode;
+import io.ballerina.openapi.cmd.Filter;
+import io.ballerina.openapi.exception.BallerinaOpenApiException;
+import io.swagger.v3.oas.models.OpenAPI;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static io.ballerina.generators.GeneratorUtils.getBallerinaOpenApiType;
+
+/**
+ * All the tests related to the auth related code snippet generation for http or oauth 2.0 mechanisms.
+ */
+public class HttpOrOAuthTests {
```

Comment:
Better if we separate HTTP and OAuth tests into two classes?

Reply:
Resolved this by adding `HttpAuthTests` and `OAuth2Tests` classes.